### ISSUE TITLE
[java] False negative in UnsynchronizedStaticFormatter

### ISSUE THREAD

**Issue by jsotuyod** [pmd/pmd#1815]
```java
public class Foo {
    private static final SimpleDateFormat sdf = new SimpleDateFormat();

    synchronized void foo() {
        sdf.format(); // static field is not protected by synchronization on non-static method
    }
}
```

_Originally posted by @boris-petrov in https://github.com/pmd/pmd/issues/1814#issuecomment-490541347_
